jdaviz/configs/cubeviz/plugins/slice/slice.py

```diff
@@ -95,6 +93,15 @@ def _update_data(self, x_all):
             x_all = x_all.value
 
         self._x_all = x_all
+
+        if self.wavelength == -1:
+            if len(x_all):
+                # initialize at middle of cube
+                self.slider = int(len(x_all)/2)
+            else:
+                # leave in the pre-init state and don't update the wavelength/slider
+                return
+
```

Comment:
this logic used to be in `_on_data_added`, but since we now attach to viewers during the plugin init, we need to make sure the slider defaults to the center the first time data is added (and no longer on the first time a viewer is watched).